src/Common/Commands.Common.Test/TestPowerBICmdletNoClientInitFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerBI.Common.Abstractions;
+using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.PowerBI.Commands.Common.Test
+{
+    public class TestPowerBICmdletNoClientInitFactory : PowerBICmdletInitFactory
+    {
+        public IPowerBIProfile Profile { get; set; }
+
+        public TestLogger Logger
+        {
+            get
+            {
+                if (this.LoggerFactory is TestLoggerFactory logger)
+                {
+                    return logger.Logger;
+                }
+
+                return null;
+            }
+        }
+
+        public TestPowerBICmdletNoClientInitFactory(bool setProfile, IPowerBIProfile profile = null) :
+            base(new TestLoggerFactory(), new ModuleDataStorage(), new TestAuthenticator(), new PowerBISettings())
+        {
+            if(setProfile)
```

Comment:
I think it might be helpful to add a comment explaining the intended test scenarios that this boolean parameter allows you to exercise.

Reply:
Added doc